src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -729,6 +729,21 @@ public InputStream exportContainer(String containerId)
   @Override
   public InputStream copyContainer(String containerId, String path)
       throws DockerException, InterruptedException {
+
+    final String apiVersion = version().apiVersion();
+    final int versionComparison = compareVersion(apiVersion, "1.20");
+
+    // Version below 1.20
+    if (versionComparison < 0) {
+      return doCopyContainer(containerId, path);
```

Comment:
This doesn't seem like the right design to me. I would not expect the `copyContainer` method to do different things depending on the version of docker you’re connected to. I would think there would be two different methods for the two different API endpoints: `copyContainer` that wraps `/containers/{id}/copy` and `archiveContainer` that wraps `/containers/{id}/archive`. `DockerClient.copyContainer` should be marked `@Deprecated`.